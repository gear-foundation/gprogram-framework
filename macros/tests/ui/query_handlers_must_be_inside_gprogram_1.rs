use gprogram_framework_macros::{gprogram, query_handlers};

#[gprogram]
mod gprogram {}

query_handlers!(
    fn value(value: u32) -> Result<u32, ()> {
        Ok(value)
    }
);

fn main() {}
