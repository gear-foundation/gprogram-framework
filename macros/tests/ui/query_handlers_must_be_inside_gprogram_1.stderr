error: The `query_handlers` block must be on the top level of `gprogram` module
 --> tests/ui/query_handlers_must_be_inside_gprogram_1.rs:7:5
  |
7 | /     fn value(value: u32) -> Result<u32, ()> {
8 | |         Ok(value)
9 | |     }
  | |_____^
